tests/bluetooth/bsim_bt/bsim_test_per_adv/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+if (NOT DEFINED ENV{BSIM_COMPONENTS_PATH})
+	message(FATAL_ERROR "This test requires the BabbleSim simulator. Please set\
+ the  environment variable BSIM_COMPONENTS_PATH to point to its components \
+ folder. More information can be found in\
+ https://babblesim.github.io/folder_structure_and_env.html")
+endif()
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(bsim_test_audio)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources} )
```

Comment:
```suggestion
target_sources(app PRIVATE
    src/common.c
    src/main.c
    src/per_adv_advertiser.c
    src/per_adv_syncer.c
)
```

Reply:
I agree with @theob-pro. We should discuss this and have the project take a definitive stance on this. @carlescufi Can you table this with the TSC maybe?

CMake official documentation discourages using FILE(GLOB) in general. The problem is that the incremental builds will not pick up new/deleted files. There is a new feature FILE(GLOB CONFIGURE_DEPENDS) that fixes this. But the documentation says it is not supported for all generators and cautions about the performance implication of scanning many directories for each incremental build.

Reply:
> CMake official documentation discourages using FILE(GLOB) in general

Can you give a reference for this? I'd like to read the arguments for it. 

I don't mind changing the code :) Reason why I ended up with `GLOB` is because I just copied this from another BSIM test

Reply:
Forgot the reference, sorry :)

> Note: We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

From https://cmake.org/cmake/help/latest/command/file.html#filesystem

Reply:
Thanks! :)